continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -175,10 +161,20 @@ object DefDefTransforms extends Trees:
           soft = false)
       )
 
+    val substituteContinuation = new TreeTypeMap(
+      treeMap = {
+        case Trees.Inlined(call, _, _)
+            if call.denot.matches(suspendContinuationMethod.symbol) =>
```

Comment:
maybe now we could use [this](https://github.com/47deg/TBD/blob/compiler-plugin/continuationsPlugin/src/main/scala/continuations/TreesChecks.scala#L9) one to avoid code  duplication?

Reply:
ah yes you are right that is a good point. feel free to ignore my comment. maybe `treeIsContinuationsSuspendContinuation` could be exported as a def in `TreesChecks`?

I think at some point we will need something like this (it is not checked that it works) but can wait for now
```
  private[continuations] def findSuspendSubtrees(tree: Tree)(using Context): List[Tree] =
    val treeIsContinuationsSuspendContinuation: Context ?=> Tree => Boolean = t =>
      t.denot.matches(suspendContinuationMethod.symbol)

    def suspendSubtrees(tree: Tree): List[Tree] =
      tree.filterSubTrees {
        case Inlined(call, _, _) => treeIsContinuationsSuspendContinuation(call)
        case t => treeIsContinuationsSuspendContinuation(t)
      }

    tree match
      case Trees.Inlined(call, _, _) => suspendSubtrees(call)
      case t => suspendSubtrees(t)
```

Reply:
> maybe treeIsContinuationsSuspendContinuation could be exported as a def in TreesChecks?
👍🏻

Reply:
I pulled this out in an extractor with tests incoming on the dependent calc branch. We can leave it for now and subsume it there when that gets in. I'm glad to have TMTOWTDI and consolidate in further prs for now.